jetstream/cli.py

```diff
@@ -533,17 +533,40 @@ def validate_config(path):
     for file in config_files:
         click.echo(f"Validate {file}", err=False)
 
-        custom_spec = AnalysisSpec.from_dict(toml.load(file))
+        if Path(file).parent.name == OUTCOMES_DIR:
+            # file is an outcome snippet
+            outcomes_spec = OutcomeSpec.from_dict(toml.load(file))
+            dummy_experiment = Experiment(
+                experimenter_slug="dummy-experiment",
+                normandy_slug="dummy_experiment",
+                type="v6",
+                status="Live",
+                branches=[],
+                probe_sets=[],
+                end_date=None,
+                reference_branch="control",
+                is_high_population=False,
+                start_date=datetime.now(),
+                proposed_enrollment=14,
+                app_id="firefox-desktop",
+                app_name="firefox_desktop",
```

Comment:
That is an interesting question. 

I suppose most outcomes are application specific (e.g. desktop-only, Fenix-only, ...) and we might want to show only outcomes as options in Experimenter that can be applied to the experiment. For example, when a user creates a new Fenix experiment in Experimenter and they are interested in configuring some outcomes, only outcomes that can actually be applied to Fenix experiments should be shown as options. Does that make sense?

Perhaps part of the outcome snippet definition is a list of `app_id`s the outcome can be computed for?

Then running the validation here would just be a matter of pasting in the right app IDs and names.

Reply:
> I suppose most outcomes are application specific (e.g. desktop-only, Fenix-only, ...) and we might want to show only outcomes as options in Experimenter that can be applied to the experiment. For example, when a user creates a new Fenix experiment in Experimenter and they are interested in configuring some outcomes, only outcomes that can actually be applied to Fenix experiments should be shown as options. Does that make sense?

Yes, that all sounds right to me.

I would _like_ binding to app_names (vs app_id) to be reasonable but it complicates validation a little bit in the Glean world since different channels have different schemas, ugh—is that actually true, or only theoretically true?—so we'd get query errors if we accidentally apply an outcome that's still riding the trains up from nightly to a release experiment.

Reply:
I was looking at https://github.com/mozilla/probe-scraper/blob/master/repositories.yaml and the thing I was concerned about was mapping app_names to the dataset, which seems only be possible using the app_id.

Reply:
Yeah, I was thinking that we could (would have to) rely on repositories.yaml for that mapping.